pkg/cmd/taskrun/log_reader.go

```diff
@@ -227,3 +235,45 @@ func getSteps(pod *corev1.Pod) []*step {
 
 	return steps
 }
+
+// reading of logs should wait untill the name of pod is
+// updates in the status. Open a watch channel on run
+// and keep checking the status until the pod name updatea
+// or the timeout is reached
+func (lr *LogReader) waitUntilPodNameAvailable(timeout time.Duration) (*v1alpha1.TaskRun, error) {
+	var first = true
+	opts := metav1.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("metadata.name", lr.Run).String(),
+	}
+	tkn := lr.Clients.Tekton
+	run, err := tkn.TektonV1alpha1().TaskRuns(lr.Ns).Get(lr.Run, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if run.Status.PodName != "" {
+		return run, nil
+	}
+
+	watchRun, err := tkn.TektonV1alpha1().TaskRuns(lr.Ns).Watch(opts)
+	if err != nil {
+		return nil, err
+	}
+	for {
+		select {
+		case event := <-watchRun.ResultChan():
+			temprun := event.Object.(*v1alpha1.TaskRun)
+			if temprun.Status.PodName != "" {
+				watchRun.Stop()
+				return temprun, nil
+			}
+			if first {
+				first = false
+				fmt.Fprintln(lr.Stream.Out, "Task still running ...")
+			}
+		case <-time.After(timeout * time.Second):
+			watchRun.Stop()
+			return nil, fmt.Errorf("No logs found")
```

Comment:
Could `No logs found` change to some message which hint there is something wrong happen, for example "Task not create success" or "Pod for task not found".

Reply:
I have updated the PR to address your comment